gen/templates/resource.go.tmpl

```diff
@@ -1412,14 +1412,9 @@ func (r *{{.ResourceClassName}}Resource) Schema(ctx context.Context, req resourc
 							stringplanmodifier.UseStateForUnknown(),
 						},
 						{{- end}}
-						{{- if or .ValidValues $.HasReadOnlyProperties}}
+						{{- if or .ValidValues}}
```

Comment:
Wouldn't this change bring the "update in place" issue we spoke about earlier? or was this fixed in Terraform side?

Reply:
I do not see this issues so not sure this is the same situation you had. 

The “MakeSingleNestedAttributeRequiredAttributesNotProvidedValidator("{{.ResourceName}}", []string{"{{listToString .RequiredPropertiesNames}}"}),”  part already handles the required setting in single nested attributes to allow for {} to be passed in when there are required attributes so the MakeStringRequired should not be used in the validator.